### ISSUE TITLE
build issue on M1 mac with pip installed cmake

### ISSUE THREAD

**Issue by HasKha** [scikit-build/cmake-python-distributions#299]
Environment: M1 mac, cmake 3.25 installed, clean conda environment with python3.10. 
Repro steps: clone https://github.com/wjakob/nanobind_example, go there and `pip install . -v` (this is not an issue with nanobind, more on this later).
Error:
```
Building wheels for collected packages: nanobind-example
  Running command Building wheel for nanobind-example (pyproject.toml)
    File "/private/var/folders/wj/f2mm6rtn1j7dl_3jwv527rrw0000gq/T/pip-build-env-6qc90q4s/overlay/lib/python3.10/site-packages/skbuild/setuptools_wrap.py", line 613, in setup
      cmkr = cmaker.CMaker(cmake_executable)
    File "/private/var/folders/wj/f2mm6rtn1j7dl_3jwv527rrw0000gq/T/pip-build-env-6qc90q4s/overlay/lib/python3.10/site-packages/skbuild/cmaker.py", line 141, in __init__
      self.cmake_version = get_cmake_version(self.cmake_executable)
    File "/private/var/folders/wj/f2mm6rtn1j7dl_3jwv527rrw0000gq/T/pip-build-env-6qc90q4s/overlay/lib/python3.10/site-packages/skbuild/cmaker.py", line 95, in get_cmake_version
      raise SKBuildError(
  Traceback (most recent call last):

  Problem with the CMake installation, aborting build. CMake executable is cmake
  error: subprocess-exited-with-error
  
  × Building wheel for nanobind-example (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> See above for output.
```

After some testing, I found out that removing `"cmake>=3.24",` from `pyproject.toml` fixes the issue. It defaults to using the system cmake and builds correctly. 

So here is another way to reproduce the bug (and why this is not a nanobind issue):
```
pip install cmake
pip install scikit-build
python

from skbuild.cmaker import *
get_cmake_version()
```
^ Same error as before

Instead:
```
pip uninstall cmake
pip install scikit-build
python

from skbuild.cmaker import *
get_cmake_version()
```
^ prints cmake version, no error

**Comment by HasKha** [comment]
this might be related to #192, the file and error message is the same, but the platform and repro steps are different, so I opened another issue.

**Comment by pfw** [comment]
I ran across this error and in my case it was cmake laying out a CMake.app app structure and downloading a cmake binary into that which MacOS flagged as a security issue and so refused to run it. `pip -vvv` gave hints to where it was building the CMake.app bundle - it would be much cleaner if it just looked for a cmake on the PATH before downloading anything.

**Comment by henryiii** [comment]
Both should be fixed. Though the "look for cmake on path" is something the build system should do, not this package. Scikit-build-core (the experimental next-generation scikit-build builder) does this correctly. However, the security issue seems to be affecting Intel too on macOS 13 (#301).

**Comment by henryiii** [comment]
By the way, we should be shipping wheels for M1, so for @HasKha, what version of Pip are you using?

**Comment by HasKha** [comment]
`pip 22.2.2 from /opt/miniconda3/envs/ren/lib/python3.10/site-packages/pip (python 3.10)`, 

from `conda install python=3.10`

**Comment by henryiii** [comment]
The app bundle issue was fixed in #304.

**Comment by HasKha** [comment]
Thank you! That fixed my issue, I'm now using the last release `cmake>=3.24.2`
